Userland/Libraries/LibC/time.cpp

```diff
@@ -147,30 +147,29 @@ static struct tm* time_to_tm(struct tm* tm, time_t t, StringView time_zone)
         t += offset->seconds;
     }
 
-    int year = 1970;
-    for (; t >= days_in_year(year) * __seconds_per_day; ++year)
-        t -= days_in_year(year) * __seconds_per_day;
-    for (; t < 0; --year)
-        t += days_in_year(year - 1) * __seconds_per_day;
+    i64 day_offset_to_epoch = t / __seconds_per_day;
+    auto [year, month, day] = days_to_time(day_offset_to_epoch);
+
     tm->tm_year = year - 1900;
+    tm->tm_yday = day_of_year(year, month, day);
 
-    VERIFY(t >= 0);
-    int days = t / __seconds_per_day;
-    tm->tm_yday = days;
-    int remaining = t % __seconds_per_day;
-    tm->tm_sec = remaining % 60;
-    remaining /= 60;
-    tm->tm_min = remaining % 60;
-    tm->tm_hour = remaining / 60;
-
-    int month;
-    for (month = 1; month < 12 && days >= days_in_month(year, month); ++month)
-        days -= days_in_month(year, month);
-
-    tm->tm_mday = days + 1;
+    tm->tm_mday = day;
     tm->tm_wday = day_of_week(year, month, tm->tm_mday);
     tm->tm_mon = month - 1;
 
+    if (t < 0) {
```

Comment:
Is this just `t = mod(t, __seconds_per_day` with

```
        auto mod = [](int a, int b) {
            return (a % b + b) % b;
        };
```

(i.e. math mod instead of C mod)?